dozer-core/src/checkpoint/mod.rs

```diff
@@ -301,13 +301,18 @@ async fn read_record_store_slices(
     let record_store = ProcessorRecordStoreDeserializer::new(record_store)?;
     let record_store_prefix = record_store_prefix(factory_prefix);
 
-    let mut last_checkpoint: Option<Checkpoint> = None;
+    let last_checkpoint: Option<Checkpoint> = None;
     let mut continuation_token = None;
     loop {
         let objects = storage
             .list_objects(record_store_prefix.to_string(), continuation_token)
             .await?;
 
+        if objects.objects.last().is_some() {
+            return Err(ExecutionError::CannotRestart);
+        }
+
+        #[cfg(FIXME_CHECKPOINTING)]
```

Comment:
We might still write the checkpoints, which means this will fail to restart every time

Reply:
Yes, we need to fix the restart issue separately. Once we settle on a solution, I will open a pull request for it.

Reply:
I think I'd prefer just solve the restart issue in this pull request as well.